aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,345 @@
+        **kwargs,
+    ) -> Message:
+        """
+        Quickly create an incoming message with the provided attributes for a dialogue.
+
+        For any attribute not provided, the corresponding default value in message is used.
+
+        :param dialogue: the dialogue to which the incoming message is intended
+        :param target: the target
+        :param performative: the performative
+        :param kwargs: other attributes
+
+        :return: the created incoming message
+        """
+        if dialogue is None:
+            raise AEAEnforceError("dialogue cannot be None.")
+
+        if dialogue.last_message is None:
+            raise AEAEnforceError("dialogue cannot be empty.")
+
+        if target is None:
+            target = dialogue.last_message.message_id
+
+        incoming_message = self.build_incoming_message(
+            message_type=dialogue._message_class,  # pylint: disable=protected-access
+            performative=performative,
+            dialogue_reference=dialogue.dialogue_label.dialogue_reference,
+            message_id=dialogue.last_message.message_id + 1,
+            target=target,
+            to=self.skill.skill_context.agent_address,
+            sender=dialogue.dialogue_label.dialogue_opponent_addr,
+            **kwargs,
+        )
+
+        return incoming_message
+
+    @staticmethod
+    def _complete_compact_message(
+        message: Tuple[Any, ...], last_incoming: Optional[bool], message_id: int
+    ) -> Tuple[Any, ...]:
+        new_message = message
+        if type(message[0]) != bool:  # pylint: disable=unidiomatic-typecheck
+            new_message = (
+                (False, *message) if last_incoming is True else (True, *message)
+            )
+        if type(new_message[1]) != int:  # pylint: disable=unidiomatic-typecheck
+            new_message_list = list(new_message)
+            new_message_list.insert(1, message_id - 1)
+            new_message = tuple(new_message_list)
+        return new_message
+
+    def prepare_dialogue(
+        self,
+        dialogues: Dialogues,
+        messages: Tuple[Tuple[Any, ...]],
+        counterparty: Address = "counterparty",
+    ) -> Dialogue:
+        """
+        Quickly create a dialogue.
+
+        The 'messages' argument is a tuple of "compact messages".
+        Each compact message is represented as a tuple (inc, target, performative, contents), where
+         - 'inc' is boolean, representing whether the message is incoming (True), or outgoing (False).
+            - 'inc' is optional, if not provided: for the first message is assumed False (outgoing),
+            for any other message, it is the opposite of the one preceding it.
+         - 'target' is integer, representing the target of the message.
+            - 'target' is optional, if not provided: for the first message is assumed 0,
+            for any other message, it is the index of the message before it in the tuple of messages + 1.
```

Comment:
All these rules around compact message sequences need to be validated somewhere and this logic needs to be implemented elsewhere to keep concerns separated.

Reply:
Done